### PR TITLE
Add logical replication test to exercise snapfiles

### PR DESCRIPTION
**Pull request by save-buffer**
## Problem
Want Test 4 from https://github.com/neondatabase/cloud/issues/13390
## Summary of changes
Adds the test

### CODE REVIEW

File: test_runner/performance/test_logical_replication.py [hunk 1]
```diff
@@ -340,3 +340,104 @@ def test_publisher_restart(
...
+                time.sleep(test_interval_min * 60)
+
+                with psycopg2.connect(connstr) as conn:
+                    with conn.cursor() as cur:
+                        cur.execute(
+                            """
+                        select
+                        count(*)
+                        from
+                            (select pg_export_snapshot() from generate_series(1, 100000) g)
```
Comment (ololobus) [inline test_runner/performance/test_logical_replication.py:425]: Is it what generates a lot of snap files? I'm not familiar enough with LR internals, so I wonder why it's only the case when there is an active LR?

Comment (hlinnaka) [reply test_runner/performance/test_logical_replication.py:425]: Hmm , I believe these are different kind of snapshot files. Logical replication creates "snapshot files" under `pg_snapshots`, and pg_export_snapshot() creates files under `pg_logical/snapshots/`.

Comment (save-buffer) [reply test_runner/performance/test_logical_replication.py:425]: This stackoverflow seems to say it stores it in `pg_snapshots`: https://stackoverflow.com/questions/73371963/what-the-pg-snapshots-folder-is-used-for

Comment (save-buffer) [reply test_runner/performance/test_logical_replication.py]: `select count(*) from pg_ls_logicalsnapdir();` returns 209 right now, so I am creating SOME of these snapshot files now

Comment (save-buffer) [reply test_runner/performance/test_logical_replication.py]: Ok we hvae it figured out

**Comment by save-buffer** [comment]
This test now requires superuser - I added an assertion to the test for that

**Review by hlinnaka — APPROVED** [review]

**Pull request merged**